src/moonlink_datafusion/src/connection_pool.rs

```diff
@@ -0,0 +1,120 @@
+use std::{collections::HashMap, sync::LazyLock};
+use tokio::{net::UnixStream, sync::Mutex};
+
+#[derive(Debug)]
+pub struct PooledStream {
+    pub uri: String,
+    pub stream: UnixStream,
+}
+
+impl PooledStream {
+    pub fn new(uri: String, stream: UnixStream) -> Self {
+        PooledStream { uri, stream }
+    }
+}
+
+static POOL: LazyLock<Mutex<HashMap<String, Mutex<Vec<UnixStream>>>>> =
+    LazyLock::new(|| Mutex::new(HashMap::new()));
+
+pub(crate) async fn get_stream(uri: &str) -> crate::Result<PooledStream> {
```

Comment:
Do you think it possible to provide a RAII wrapper, which gets or create a connection on construction, and returns back on destruction? That way we never forget return.